stdlib/date/truncate_duration_test.flux

```diff
@@ -33,12 +45,38 @@ outData =
 ,,0,FF,_m,2018-05-22T19:53:40.000000000Z,1893459600000000000
 ,,0,FF,_m,2018-05-22T19:53:50.000000000Z,1893463200000000000
 ,,0,FF,_m,2018-05-22T19:54:00.000000000Z,1893466800000000000
-"
-t_duration_truncate = (table=<-) =>
-    table
-        |> range(start: 2018-05-22T19:53:00Z)
-        |> drop(columns: ["_start", "_stop"])
-        |> map(fn: (r) => ({r with _value: int(v: date.truncate(t: duration(v: r._value), unit: 1s))}))
-
-test _duration_truncate = () =>
-    ({input: testing.loadStorage(csv: inData), want: testing.loadMem(csv: outData), fn: t_duration_truncate})
+",
+            )
+
+        testing.diff(got: got, want: want)
+    }
+
+testcase truncate_duration_location {
+        option location = timezone.location(name: "Europe/Madrid")
+
+        got =
+            csv.from(csv: inData)
+                |> range(start: 2018-05-22T19:53:00Z)
+                |> drop(columns: ["_start", "_stop"])
+                |> map(fn: (r) => ({r with _value: int(v: date.truncate(t: duration(v: r._value), unit: 1mo))}))
+
+        want =
+            csv.from(
+                csv:
+                    "
+#datatype,string,long,string,string,dateTime:RFC3339,long
+#group,false,false,true,true,false,false
+#default,_result,,,,,
+,result,table,_field,_measurement,_time,_value
+,,0,FF,_m,2018-05-22T19:53:00.000000000Z,1893452400000000000
+,,0,FF,_m,2018-05-22T19:53:10.000000000Z,1890774000000000000
+,,0,FF,_m,2018-05-22T19:53:20.000000000Z,1890774000000000000
+,,0,FF,_m,2018-05-22T19:53:30.000000000Z,1893452400000000000
+,,0,FF,_m,2018-05-22T19:53:40.000000000Z,1893452400000000000
+,,0,FF,_m,2018-05-22T19:53:50.000000000Z,1893452400000000000
+,,0,FF,_m,2018-05-22T19:54:00.000000000Z,1893452400000000000
+",
+            )
+
+        testing.diff(got: got, want: want)
+    }
```

Comment:
Nit: Missing newline at the end of file (usually your editor would add one in my experience :shrug: )

Reply:
I will address this too 👍

Reply:
> usually your editor would add one in my experience 

Not for unknown filetypes (at least in Golang :) )